artichoke-backend/src/extn/core/string/trampoline.rs

```diff
@@ -42,31 +44,324 @@ pub fn add(interp: &mut Artichoke, mut value: Value, mut other: Value) -> Result
 
     let mut concatenated = s.clone();
     // XXX: This call doesn't do a check to see if we'll exceed the max allocation
-    //    size and may panic.
+    //    size and may panic or abort.
     concatenated.extend_from_slice(to_append);
     super::String::alloc_value(concatenated, interp)
 }
 
 pub fn push(interp: &mut Artichoke, mut value: Value, mut other: Value) -> Result<Value, Error> {
+    if value.is_frozen(interp) {
+        let s = unsafe { super::String::unbox_from_value(&mut value, interp)? };
+        let message = "can't modify frozen String: "
+            .chars()
+            .chain(s.inspect())
+            .collect::<super::String>();
+        return Err(FrozenError::from(message.into_vec()).into());
+    }
+
     let mut s = unsafe { super::String::unbox_from_value(&mut value, interp)? };
+    if let Ok(int) = other.try_convert_into::<i64>(interp) {
+        return match s.encoding() {
```

Comment:
How do you feel about `push_int` in `spinoso-string` and letting the encoded types to deal with this?

Reply:
that's a good idea! let me file a ticket.

Edit: done!

- https://github.com/artichoke/artichoke/issues/1981